pkg/execution/executor/executor.go

```diff
@@ -1846,16 +1857,6 @@ func (e *executor) Resume(ctx context.Context, pause state.Pause, r execution.Re
 	}
 
 	err = util.Crit(ctx, "consume pause", func(ctx context.Context) error {
-		// Lease this pause so that only this thread can schedule the execution.
-		//
-		// If we don't do this, there's a chance that two concurrent runners
-		// attempt to enqueue the next step of the workflow.
-		err = e.pm.LeasePause(ctx, pause.ID)
-		if err == state.ErrPauseLeased || err == state.ErrPauseNotFound {
-			// Ignore;  this is being handled by another runner.
-			return nil
-		}
```

Comment:
no longer leasing?

Reply:
correct.  the entire code is idempotent:

- consuming a pause and saving step details is transactional and idempotent, and doesn't conflict if two concurrent threads attempt to save the same step
- and given that, only one thread can schedule the next step.

leasing is no longer necessary.